app/controllers/graphql_controller.rb

```diff
@@ -0,0 +1,43 @@
+class GraphqlController < ApplicationController
+  def execute
+    variables = ensure_hash(params[:variables])
+    query = params[:query]
+    operation_name = params[:operationName]
+    context = {
+      # Query context goes here, for example:
+      # current_user: current_user,
+    }
+    result = ApprenticeProgressAPISchema.execute(query, variables: variables, context: context, operation_name: operation_name)
+    render json: result
+  rescue => e
+    raise e unless Rails.env.development?
+    handle_error_in_development e
+  end
+
+  private
+
+  # Handle form data, JSON body, or a blank value
+  def ensure_hash(ambiguous_param)
```

Comment:
Thoughts on extracting this to a class, which could be tested in isolation?

Reply:
Sorry, should have made a comment in the PR. This file is graphql boilerplate which I haven't touched. So I'll keep as is for the moment until I need to use it.